pkg/endpoint/bpf.go

```diff
@@ -955,9 +955,9 @@ func (e *Endpoint) finalizeProxyState(regenContext *regenerationContext, err err
 		// Always execute the finalization code, even if the endpoint is
 		// terminating, in order to properly release resources.
 		e.unconditionalLock()
+		defer e.unlock() // In case Finalize() panics
 		e.getLogger().Debug("Finalizing successful endpoint regeneration")
 		datapathRegenCtx.finalizeList.Finalize()
-		e.unlock()
```

Comment:
Doesn't e`.unlock()` following `Revert()` below have the same issue?

Reply:
Good catch - it does.

Reply:
Fixed.